UserService/src/main/java/com/streamster/userservice/controller/UserController.java

```diff
@@ -1,28 +1,39 @@
 package com.streamster.userservice.controller;
 
+import com.streamster.userservice.model.SystemRoleType;
 import com.streamster.userservice.model.User;
-import com.streamster.userservice.model.dto.UserInfo;
-import com.streamster.userservice.repository.UserRepository;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import com.streamster.userservice.model.dto.RegistrationDTO;
+import com.streamster.userservice.service.UserService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
 
 @RestController
 @RequestMapping("/user")
 public class UserController {
 
-    final UserRepository repository;
+    final UserService userService;
 
-    public UserController(UserRepository repository) {
-        this.repository = repository;
+    public UserController(UserService userService) {
+        this.userService = userService;
     }
 
-    @PostMapping("/register")
-    void register(@RequestBody UserInfo userInfo) {
-        userInfo = userInfo.encryptPassword();
-        repository.save(User.fromUserInfo(userInfo));
+    @GetMapping("/test")
+    ResponseEntity<String> test() {
+        return new ResponseEntity<>("SUCCESS",HttpStatus.OK);
     }
 
+    @PostMapping("/register")
+    ResponseEntity<String> register(@RequestBody RegistrationDTO registrationDTO) {
+        userService.register(User.fromRegistrationDTO(registrationDTO.encryptPassword()));
+        return new ResponseEntity<>("The account was created successfully", HttpStatus.CREATED);
+    }
 
+    @PreAuthorize("hasAuthority(T(com.streamster.userservice.model.SystemRoleType).ADMIN)")
+    @PutMapping("/updateSystemRole/{userId}/{role}")
```

Comment:
I'd rather use RequestParam for role. In this case user is a resource and you want to update its attribute role. And the path should be to the resource